libs/newsletters-data-client/src/lib/newsletter-storage/NewsletterStorage.ts

```diff
@@ -42,9 +42,7 @@ export abstract class NewsletterStorage {
 
 	abstract delete(
 		listId: number,
-	): Promise<
-		SuccessfulStorageResponse<NewsletterData> | UnsuccessfulStorageResponse
-	>;
+	): Promise<SuccessfulStorageResponse<string> | UnsuccessfulStorageResponse>;
```

Comment:
I get that fetching the data from s3 before deleting it is a pain, so returning a copy of the deleted `NewsletterData` is probably overkill - but just returning a string might make it harder for the consumer of the storage to interpret and react to  the responses - I'd tend to return a simple object like:
```
type DeletionConfirmation = {
  message: string;
  listId: number;
}
```